src/java/org/apache/cassandra/concurrent/Stage.java

```diff
@@ -152,6 +167,18 @@ public static void shutdownNow()
         ExecutorUtils.shutdownNow(executors());
     }
 
+    public static void shutdownAndAwaitMutatingExecutors(boolean interrupt, long timeout, TimeUnit units) throws InterruptedException, TimeoutException
+    {
+        List<ExecutorPlus> executors = mutatingExecutors();
+        ExecutorUtils.shutdown(interrupt, executors);
+        ExecutorUtils.awaitTermination(timeout, units, executors);
```

Comment:
Does `ExecutorUtils.awaitTermination()` fail all the termination awaits if the very first one times out? I guess that might not be a terrible thing, and we'll at lest have something visibly exploding, but wanted to make sure that was cool.

Reply:
Yes, `ExecutorUtils.awaitTermination()` will stop waiting for the other executors as soon as it hits a timeout awaiting. Not an issue in this case, but it is interesting that if there was an exception before waiting for a {{Thread}}, it would never be joined.